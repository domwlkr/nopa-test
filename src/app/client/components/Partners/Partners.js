import React from 'react';

const partners = [{
    name: 'Airbnb',
    logo: require('../../../static/images/Airbnb.png')
  },
  {
    name: 'Metro',
    logo: require('../../../static/images/Metro.png')
  },
  {
    name: 'Pariti',
    logo: require('../../../static/images/Pariti.png')
  },
  {
    name: 'Unshackled',
    logo: require('../../../static/images/Unshackled.png')
  }
];

const partnerLogos = partners.map((partner) =>
    <div className="partner" key={partner.name}><img alt={partner.name} src={partner.logo} /></div>
);

const Partners = (props) => {
  return (
    <section className="partners">
      <div>
        <div>
          <span>Our partners:</span>
        </div>
        <div>
          {partnerLogos}
        </div>
      </div>
    </section>
  );
};

export default Partners;
