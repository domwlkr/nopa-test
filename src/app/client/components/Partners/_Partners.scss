.partners {
  background-color: #2ABBF0;
  display: flex;
  justify-content: center;
  text-align: center;
  color: white;
  padding: 70px 5%;

  > div {
    display: flex;
    flex-wrap: wrap;
    flex-direction: column;
    max-width: 846px;
    margin: 0 auto;

    > div:nth-child(1) {
      margin-bottom: 30px;
    }
  }

  .partner {
    margin-bottom: 55px;

    &:last-of-type {
      margin-bottom: 0;
    }
  }
}

@media (min-width:380px) {
  .partners {
    flex-wrap: nowrap;
    text-align: left;

    > div {
      flex-wrap: nowrap;
      flex-direction: row;
      align-items: flex-start;
      width: 100%;

      > div:nth-child(1) {
        flex: 1;
        margin-bottom: 0;
      }

      > div:nth-child(2) {
        display: flex;
        flex-wrap: nowrap;
        justify-content: space-between;
        flex: 4;
      }
    }

    .partner {
      margin-bottom: 0;
    }
  }
}
