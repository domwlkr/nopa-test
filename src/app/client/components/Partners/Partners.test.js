import React from 'react';
import Partners from './Partners';
import renderer from 'react-test-renderer';

test('Renders correctly', () => {
  const partners = renderer.create(
    <Partners />
  ).toJSON();

  expect(partners).toMatchSnapshot();
});