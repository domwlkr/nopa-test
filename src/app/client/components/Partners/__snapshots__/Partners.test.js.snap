// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<section
  className="partners"
>
  <div>
    <div>
      <span>
        Our partners:
      </span>
    </div>
    <div>
      <div
        className="partner"
      >
        <img
          alt="Airbnb"
          src="test-file-stub"
        />
      </div>
      <div
        className="partner"
      >
        <img
          alt="Metro"
          src="test-file-stub"
        />
      </div>
      <div
        className="partner"
      >
        <img
          alt="Pariti"
          src="test-file-stub"
        />
      </div>
      <div
        className="partner"
      >
        <img
          alt="Unshackled"
          src="test-file-stub"
        />
      </div>
    </div>
  </div>
</section>
`;
